overlord/snapstate/handlers.go

```diff
@@ -2756,6 +2794,10 @@ func (m *SnapManager) doClearSnapData(t *state.Task, _ *tomb.Tomb) error {
 			return err
 		}
 
+		if err = m.backend.RemoveSnapSaveData(info, dirOpts); err != nil {
```

Comment:
It felt natural to have the removal of snap save data as a part of doClearSnapData, and that avoided having a seperate removal step to handle this. Unfortunately there was no natural place to do the opposite of this, and I ended up feeling better about having a seperate step during install to create the snap save data directory.